src/with-urql-client.tsx

```diff
@@ -35,7 +38,7 @@ export interface NextUrqlContext extends NextPageContext {
 
 type NextUrqlClientConfig =
   | NextUrqlClientOptions
-  | ((ctx: NextPageContext) => NextUrqlClientOptions);
+  | ((ctx?: NextPageContext) => NextUrqlClientOptions);
```

Comment:
`ctx` is now potentially `undefined` in the case of the first render **on the client-side**. Our types need to guard against that case. I'm not sure this is the best way to express potentially `undefined` here, or if we need to do something like `(ctx: NextPageContext | undefined)` to be more explicit.